src/robot/ConcreteRobot.cpp

```diff
@@ -331,6 +331,64 @@ ConcreteRobot::getTrajectoryPostProcessor(
       feasibilityApproxTolerance);
 }
 
+//==============================================================================
+std::shared_ptr<aikido::planner::TrajectoryPostProcessor>
+ConcreteRobot::getTrajectoryPostProcessor(
+    const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+    const aikido::planner::PostProcessorParams& postProcessorParams) const
+{
+  // TODO (sniyaz):Should we  delete `smoothPath`, `retimePath`, and
+  // `retimePathWithKunz` in a subsequent PR?
+  // TODO: (sniyaz): Include custom limits in params.
+  switch (postProcessorParams.mPostProcessorType)
+  {
+    case aikido::planner::PostProcessorType::HAUSER:
+      return std::make_shared<ParabolicSmoother>(
+          getVelocityLimits(*metaSkeleton),
+          getAccelerationLimits(*metaSkeleton),
+          postProcessorParams.mHauserParams.mEnableShortcut,
+          postProcessorParams.mHauserParams.mEnableBlend,
+          postProcessorParams.mHauserParams.mShortcutTimelimit,
+          postProcessorParams.mHauserParams.mBlendRadius,
+          postProcessorParams.mHauserParams.mBlendIterations,
+          postProcessorParams.mHauserParams.mFeasibilityCheckResolution,
+          postProcessorParams.mHauserParams.mFeasibilityApproxTolerance);
+    case aikido::planner::PostProcessorType::KUNZ:
+      return std::make_shared<KunzRetimer>(
+          getVelocityLimits(*metaSkeleton),
+          getAccelerationLimits(*metaSkeleton),
+          postProcessorParams.mKunzParams.mMaxDeviation,
+          postProcessorParams.mKunzParams.mTimeStep);
+    default:
+      // TODO (sniyaz): Can this be made a compile-time check instead?
+      std::stringstream message;
+      message << "PostProcessor type not recognized in `postProcessorParams`!";
+      throw std::runtime_error(message.str());
```

Comment:
@brianhou is throwing a runtime_error normal at this stage? What would be wrong with returning a `nullptr` and printing the message to stderr?

Reply:
If I remember correctly, we opted to be super strict here because we didn't have a compile-time check but really wanted one. (And that this would be something ironed out by testing in sim anyway before it goofs up on hardware.) Returning a `nullptr` in both places seems reasonable.

Reply:
I don't think we need to throw anything here, right? Since now we have nice templating that checks on its own that the types match up.